add OpenBSD to XInitThreads() code path to make calls to X11 threadsafe

Index: rts/System/SpringApp.cpp
--- rts/System/SpringApp.cpp.orig
+++ rts/System/SpringApp.cpp
@@ -296,7 +296,7 @@ bool SpringApp::Init()
 
 bool SpringApp::InitPlatformLibs()
 {
-#if !(defined(_WIN32) || defined(__APPLE__) || defined(HEADLESS))
+#if !(defined(_WIN32) || defined(__APPLE__) || defined(HEADLESS)) || defined(__OpenBSD__)
 	// MUST run before any other X11 call (including
 	// those by SDL) to make calls to X11 threadsafe
 	if (!XInitThreads()) {
